garden_ai/gardens.py

```diff
@@ -142,6 +142,24 @@ def pipeline_names(self) -> List[str]:
             names += [name]
         return names
 
+    def _set_pipelines_from_remote_metadata(self, pipeline_metadata: list[dict]):
```

Comment:
ok more qualitatively, this does make me a little nervous bc now there's a way to have a Garden instance refer to/call a pipeline even if it doesn't actually have that pipeline's ID registered -- I think the shortest way home is to just add a check to assert that `pipeline.uuid in self.pipeline_ids`, but I wonder if this is the "`RegisteredPipeline`" moment for gardens, i.e. one kind lives locally and can be messed with, and one is the stricter representation of a published record. Maybe `PublishedGarden` instantiates from search results and `Garden` instantiates from local data?